subprojects/gradle/build-metrics/src/test/kotlin/com/avito/android/plugin/build_metrics/ConfigurationCacheCompatibilityTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.avito.android.plugin.build_metrics
+
+import com.avito.test.gradle.TestProjectGenerator
+import com.avito.test.gradle.TestResult
+import com.avito.test.gradle.gradlew
+import com.avito.test.gradle.module.AndroidAppModule
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+internal class ConfigurationCacheCompatibilityTest {
+
+    /**
+     * TODO blocked by agp 4.2
+     */
+    @Disabled
+    @Test
+    fun `configuration with applied plugin - ok`(@TempDir projectDir: File) {
+        TestProjectGenerator(
+            name = "rootapp",
+            plugins = listOf("com.avito.android.build-metrics"),
+            modules = listOf(
+                AndroidAppModule(name = "app")
+            )
+        ).generateIn(projectDir)
+
+        runTask(projectDir).assertThat().buildSuccessful()
+
+        runTask(projectDir).assertThat().buildSuccessful().configurationCachedReused()
```

Comment:
We duplicate the same pattern of the test in all plugins.
Is it possible to reuse this contract and define only specific parts?
The issue is not in code duplication but rather in the common contract.
Another example: test for caching.

Reply:
One probable option: [junit 5 extensions](https://medium.com/decisionbrain/factor-test-code-the-junit-5-way-that-is-without-inheritance-9ba2784d34)
It looks overcomplicated for such simple cases without a complex shared state.